charts/cluster-api-runtime-extensions-nutanix/templates/helm-repository.yaml

```diff
@@ -0,0 +1,78 @@
+# Copyright 2024 Nutanix. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+# This file contains the manifests to run a helmRepository deployment which contains helm charts for our addons.
+# The pod is built via goreleaser with configuration from hack/addons.
+#
+{{ if .Values.selfHostedRegistry }}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: helm-repository-selfsigned
+  namespace: {{ .Release.Namespace }}
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: helm-repository-tls
+  namespace: {{ .Release.Namespace }}
+spec:
+  dnsNames:
+  - helm-repository.{{ .Release.Namespace }}.svc
+  - helm-repository.{{ .Release.Namespace }}.svc.cluster.local
+  issuerRef:
+    name: helm-repository-selfsigned
+    kind: Issuer
+  secretName: helm-repository-tls
+---
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
Let's add tolerations and node affinity (`preferredDuringSchedulingIgnoredDuringExecution`) to allow running on CP nodes but prefer to run on non-CP nodes.